### ISSUE TITLE
train script for improve yolo accuracy

### ISSUE THREAD

**Issue by johnSmith1990** [tau-adl/Detection_Tracking_JetsonTX2#11]
Hi. Is there any train script to retrain yolo network?
thanks a lot

**Comment by MuhammadAsadJaved** [comment]
@johnSmith1990   You can use any of these projects to train Yolov3. 

https://github.com/ultralytics/yolov3

https://github.com/AlexeyAB/darknet

**Comment by johnSmith1990** [comment]
Mashkorin @MuhammadAsadJaved,
the detection model is tiny here? or full yolo? cfg file seems to be for tiny yolo.

I am trying to run detect and track but I have to compile caffe first. But in general I am working on tracking project.
I will share my results here.

**Comment by MuhammadAsadJaved** [comment]
> Mashkorin @MuhammadAsadJaved,
> the detection model is tiny here? or full yolo? cfg file seems to be for tiny yolo.
> 
> I am trying to run detect and track but I have to compile caffe first. But in general I am working on tracking project.
> I will share my results here.

 You are right. It's Yolov3-Tiny. 

You can read this issue we are facing from several days. I also mentioned the caffe build process in the issue. 

https://github.com/tau-adl/Detection_Tracking_JetsonTX2/issues/7

The detector is detecting very well,  You can try with `python3 mainDetect.py `   
But the tracker and tracker combined with Yolo is not working. I hope you will debug the code. 
You can try only tracker with  `python3 mainGoturnTracker.py `

**Comment by johnSmith1990** [comment]
thanks. 
First of all, I tested several video for detection but detection failed in many frames. so I am trying to test full yolo for good FPS results. Then train for drone detection. Then compiling caffe and test tracker. This is my roadmap. 
I want to run my project on 1080 ti. 

Do you use full yolo or tiny version? Is full yolo real time?

**Comment by MuhammadAsadJaved** [comment]
> thanks.
> First of all, I tested several video for detection but detection failed in many frames. so I am trying to test full yolo for good FPS results. Then train for drone detection. Then compiling caffe and test tracker. This is my roadmap.
> I want to run my project on 1080 ti.
> 
> Do you use full yolo or tiny version? Is full yolo real time?

I am using same given weights. 

I think first you should try to use the original code and make sure it can work or not, if it can then you can train yolov3 to replace detector. Otherwise it will waste your time to train if tracker can not work.  So my suggestion is first we should make tracker work and then train detector. 

I just figured it out the problem is with the tracker draw function. 
https://github.com/tau-adl/Detection_Tracking_JetsonTX2/blob/master/DetectAndTrackUtils.py#L58

I just checked the x1 and x2 , y1 and y2 values was same, so tracker was not drawing bounding box.  So i just added 50 and 30 in x2 and y2 to see the effect of the bounding box.  Now it's drawing the bounding box but just after the detection ,the tracker's bounding box start moving at the bottom left.  as you can see the first picture is detection and the second picture is very next frame after detection when it start tracking.  So now I want to figure it out that why tracker's draw function is getting wrong values for bounding box. 
![f1](https://user-images.githubusercontent.com/28862708/97847144-19959f00-1d2a-11eb-812f-3a6da8e2baf6.png)
![f2](https://user-images.githubusercontent.com/28862708/97847149-1ac6cc00-1d2a-11eb-9ec7-613bc789b049.png)

**Comment by johnSmith1990** [comment]
Ok.

**Comment by johnSmith1990** [comment]
@MuhammadAsadJaved  May you help me in compiling caffe? I am using python 3.6 cuda 10,windos 10 OS.

**Comment by MuhammadAsadJaved** [comment]
@johnSmith1990  So sorry I never tried in windows os.

**Comment by johnSmith1990** [comment]
@MuhammadAsadJaved  Hello
How are you?
Do you understand how are the annotation of Alov300++ dataset?
I want to label my videos and train goturn network.
regards

**Comment by MuhammadAsadJaved** [comment]
Hi, I am good . Thanks. Sorry I haven’t checked it. I will check it out on
Monday and then let you know.

**Comment by johnSmith1990** [comment]
Hi @MuhammadAsadJaved.
I got the goturn for pytorch and run it successfully(https://github.com/nrupatunga/goturn-pytorch) and I want to combine it with yolo network.

**Comment by gzc-123456-gzc** [comment]
@johnSmith1990 
Hello friends, haven't you trained the tracking algorithm successfully?  Is the GOTURN algorithm used is the code given by the author? If it is, it means that there is no problem with the tracking algorithm code. The problem may appear in the pre-trained weight file of GOTURN.

**Comment by gzc-123456-gzc** [comment]
@johnSmith1990 Hello friends, have you successfully trained the tracking algorithm?  Is the GOTURN algorithm used? Is the code given by the author? If it is, it means that there is no problem with the tracking algorithm code. The problem may appear in the pre- trained weight file of GOTURN.Because you successfully ran the GOTURN algorithm, and another friend @MuhammadAsadJaved  successfully ran the yolov3 detection algorithm

**Comment by johnSmith1990** [comment]
Hi @gzc-123456-gzc 
I run goturn successfully on my machine but because of problem in labeling of my data for goturn training also low FPS of goturn on gpu, i decided to pause using it. I switched and I am searching other methods to customize.
